website/frontend/src/tests/FormPayload/FormPayload.test.tsx

```diff
@@ -0,0 +1,500 @@
+import { Formik, Form } from 'formik';
+import { render, screen, within, fireEvent, act, waitFor, cleanup } from '@testing-library/react';
+
+import { FormValues, TestFormProps } from './types';
+import { INPUT_PAYLOAD_VALUES, INPUT_MANUAL_PAYLOAD, INIT_FORM_VALUES, INPUT_FILE_CONTENT } from './inputData';
+
+import { FILE_TYPES } from 'consts';
+import { getPreformattedText } from 'helpers';
+import { AlertProvider } from 'context/alert';
+import { FormPayload } from 'components/common/FormPayload';
+
+const TestFromPayload = ({ values, onSubmit }: TestFormProps) => (
+  <AlertProvider>
+    <Formik initialValues={INIT_FORM_VALUES} onSubmit={onSubmit}>
+      <Form data-testid="test-form">
+        <FormPayload name="payload" values={values} />
+      </Form>
+    </Formik>
+  </AlertProvider>
+);
+
+describe('from payload tests', () => {
+  const submitCallbackMock = jest.fn();
+
+  const submit = () => {
+    submitCallbackMock.mockReset();
+    fireEvent.submit(screen.getByTestId('test-form'));
+  };
+
+  const changeFieldValue = (element: Element | Node, value: string) =>
+    act(() => {
+      fireEvent.change(element, { target: { value } });
+    });
+
+  const toggleView = () =>
+    act(() => {
+      fireEvent.click(screen.getByRole('checkbox'));
+    });
+
+  const verifyValues = (formValues: FormValues) => waitFor(() => expect(submitCallbackMock).toBeCalledWith(formValues));
+
+  afterEach(() => {
+    cleanup();
+    jest.resetAllMocks();
+  });
+
+  it('should displayed manual textarea', () => {
+    const { rerender } = render(<TestFromPayload onSubmit={jest.fn()} />);
+
+    const manualTextarea = screen.getByPlaceholderText('// Enter your payload here');
+
+    expect(screen.queryByLabelText('Manual input')).toBeNull();
+    expect(manualTextarea).toBeInTheDocument();
+    expect(manualTextarea).toHaveTextContent('');
+    expect(screen.queryByText('Select file')).toBeNull();
+
+    rerender(<TestFromPayload values={INPUT_PAYLOAD_VALUES} onSubmit={jest.fn()} />);
+
+    expect(screen.getByLabelText('Manual input')).toBeInTheDocument();
+
+    expect(manualTextarea).not.toBeInTheDocument();
+    expect(screen.getByRole('checkbox')).not.toBeChecked();
+    expect(screen.queryByText('Select file')).toBeNull();
+
+    expect(screen.getAllByRole('group')).toHaveLength(2);
+    expect(screen.getAllByRole('combobox')).toHaveLength(2);
+
+    toggleView();
+
+    expect(screen.getByText('Select file')).toBeInTheDocument();
+  });
+
+  it('should submit correct manual', async () => {
+    render(
+      <TestFromPayload
+        values={INPUT_PAYLOAD_VALUES}
+        onSubmit={jest.fn((values: FormValues) => submitCallbackMock(values))}
```

Comment:
why wrapping in `jest.fn` if it's already returning one?

Reply:
There are other arguments, not just values